### PR TITLE
[FOR REVIEW] Add zeroex settler txns on ethereum

### PR DESCRIPTION
**Pull request by RantumBits**
### Contribution type
Please check the type of contribution this pull request is for:

- [x] New spell(s)
- [ ] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name* zeroex_ethereum.settler_trades
- **Description:** [Detailed description of the new spell(s) and their purpose] decode 0x API trades on new settler contracts
- **Who are the new spell(s) for?** [Internal team or general community] Both
- **How will the new spell(s) be used downstream?** [Description of downstream usage] Merged into dex_aggregator.trades
- **Implementation details:** [Information on how the spell(s) are implemented] gets taker and maker amounts only, no fills level data for intermediary swaps
- **Test instructions:** [How to test the new spell(s)]- https://dune.com/queries/3921632
- **Related issue(s):** [Link to related issues, if any]

---

### CODE REVIEW

**Comment by RantumBits** [comment]
this is updated to match #6425

File: dbt_subprojects/dex/models/_projects/zeroex/polygon/zeroex_polygon_settler_trades.sql [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+            AND minute >= cast('{{zeroex_settler_start_date}}' as date)
+            {% endif %}
+        ), 
+results as (
+    select 
+        trades.block_time,
+        trades.block_number,
+        zid,
+        trades.contract_address,
+        methodID,
```
Comment (Hosuke) [inline dbt_subprojects/dex/models/_projects/zeroex/polygon/zeroex_polygon_settler_trades.sql:140]: Please follow the naming standard in #6425 mentioned.

File: dbt_subprojects/dex/models/_projects/zeroex/polygon/zeroex_polygon_settler_trades.sql [hunk 2]
```diff
@@ -0,0 +1,247 @@
+{{  config(
+
+        schema = 'zeroex_polygon',
+        alias = 'settler_trades',
+        materialized='incremental',
+        partition_by = ['block_month'],
+        unique_key = ['block_date', 'tx_hash', 'evt_index'],
```
Comment (Hosuke) [inline dbt_subprojects/dex/models/_projects/zeroex/polygon/zeroex_polygon_settler_trades.sql:7]:
```suggestion
        unique_key = ['block_month', 'block_date', 'tx_hash', 'evt_index'],
```

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: dbt_subprojects/dex/models/_projects/zeroex/polygon/zeroex_polygon_settler_trades.sql (hunk 2 above)
Comment (RantumBits) [reply dbt_subprojects/dex/models/_projects/zeroex/polygon/zeroex_polygon_settler_trades.sql]: thanks for pointing out. this polygon model wasn't intended to be included here. it's now removed @Hosuke

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @RantumBits 

[Sample zeroex_ethereum_settler_trades]()

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**